Source/Blazorise/Adapters/CloseableAdapter.cs

```diff
@@ -33,12 +33,6 @@ public async Task Run( bool visible )
 
                 await Task.Delay( animatedComponent.AnimationDuration );
 
-                // NOT SURE IF WE NEED THIS?
```

Comment:
@David-Moreira do you think we still need this?

Reply:
The comment? xD Or the `Task.Delay`? I think the `Task.Delay` was to give it the time to animate before proceeding with the next operation. But I can review that.

Reply:
No, the Show and Hide method calls.

Reply:
This part

```cs
if ( visible )
    await animatedComponent.Show();
else
    await animatedComponent.Hide();
```

Reply:
I no longer remember this by heart. I'll take a look.

Reply:
So, why do you think we don't need that part? That part is how it works without animation.
![image](https://user-images.githubusercontent.com/22283161/170847361-cbb00c87-cc2f-420a-96f0-d10f13b090de.png)

Reply:
That same show/hide block was included above for the animated component path after the Task. Delay.

Reply:
> So, why do you think we don't need that part? That part is how it works without animation. ![image](https://user-images.githubusercontent.com/22283161/170847361-cbb00c87-cc2f-420a-96f0-d10f13b090de.png)

Well, the comment said we

```
// NOT SURE IF WE NEED THIS?
```

so just wanted to make sure. Just going to remove the comment so that it doesn't confuse us again 🤣

Reply:
> That same show/hide block was included above for the animated component path after the Task. Delay.

AHHH I didn't get that, I just re-tested yea, you're right seems to work just the same.
But let's leave it as original for now, I think it's better if we still run it if animated. 

From an API perspective, doesn't matter if animated or not, seems like Show/Hide should be expected to run I'd guess... We might want to run some additional or centralized logic there. 
But It's true, I guess it's just the way that animation works, ends up being pretty close to `Show`, because most times you'll have to actually start showing the component to animate it... and ends up being a bit redudant... But we might get different components that need it like this in the future...

For example, `Hide` seems to have a centralized closeReason reset there.
![image](https://user-images.githubusercontent.com/22283161/170868312-8c8b1db7-6e18-4dae-9af5-f5c95dd21d2e.png)

@Bobbar 
Now I was taking a look at the original branch commit: 
> "an infinite loop within the Modal component if a Hide is called before a Show call finishes." 
Is this easily reproducable? Can you help re-create this?

Reply:
It's easily reproducible on the boostrap demo site by quickly double-clicking the basic modal button.

Reply:
I cannot reproduce it myself.

Reply:
I can reproduce it easily on the bootstrap demo site by double-clicking the standard modal button.

Reply:
@Bobbar you're right.
Looks like disco night :)

So you solved this by removing that `show/hide` block, right?

Reply:
Right. I don't recall any other components implementing the animation. And the modal calls the show/hide internally. IIRC.